Restructuring the semantic_text field type page (#138571)
elastic/elasticsearch: docs/reference/elasticsearch/mapping-reference/semantic-text.md

> # Semantic text field type [semantic-text]

Reviewer: I think the landing page introduction could be expanded a little more about the problems semantic_text solves (some top ideas could be mined from blog posts) and the manual workflows it replaces. Namely, we could state more explicitly that this enables _semantic search_.

Reviewer: - It could be nice to have the simplest possible example of a semantic_text mapping request like the one here for illustration: https://www.elastic.co/guide/en/elasticsearch/reference/8.18/semantic-search-semantic-text.html#semantic-text-index-mapping
- Relatedly, we should probably link to the semantic_search tutorial from here

Author: hey @leemthompo, based on this suggestion I expanded this part a little bit (and shortened the Overview section). LMKWYT about it now :)

# Semantic text field type [semantic-text]

The `semantic_text` field type simplifies [semantic search](docs-content://solutions/search/semantic-search.md) by providing defaults for the required infrastructure. Using `semantic_text`, you don't have to manually configure mappings, set up ingestion pipelines, or handle chunking. The field type automatically:

- Configures index mappings: Chooses the correct field type (`sparse_vector` or `dense_vector`), dimensions, similarity functions, and storage optimizations based on the inference endpoint.
- Generates embeddings during indexing: Automatically generates embeddings when you index documents, without requiring ingestion pipelines or inference processors.
- Handles chunking: Automatically chunks long text documents during indexing.

> The following example creates an index mapping with a `semantic_text` field:

Reviewer: ```suggestion
## Basic `semantic_text` mapping example

The following example creates an index mapping with a `semantic_text` field:
```

totally optional, very in the weeds here 😄, but maybe it would be nice to set the example off under its own subheading

```console
PUT semantic-embeddings 
{
  "mappings": { <1>
    "properties": {
      "content": { 
        "type": "semantic_text" 
      }
    }
  }
}
```
% TEST[skip:Requires inference endpoint]

> 1. In this example, there is no `inference_id` is specified, so `semantic_text` uses a [default inference endpoint](./semantic-text-how-tos.md#default-and-preconfigured-endpoints).

Reviewer: ```suggestion
1. In this example, the `semantic_text` field uses a [default inference endpoint](./semantic-text-how-tos.md#default-and-preconfigured-endpoints) because the `inference_id` parameter isn't specified.
```
